src/App.tsx

```diff
@@ -15,48 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SortDefault = '',
+  SortAlphabet = 'abc',
+  SortLength = 'length',
+}
+
+interface FilterParam {
+  sortFilter: SortType;
+  isReverseFilter: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortFilter, isReverseFilter } : FilterParam,
+) {
+  const prepareGoods = [...goods];
+
+  if (sortFilter) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortFilter) {
+        case (SortType.SortAlphabet):
+          return good1.localeCompare(good2);
+
+        case (SortType.SortLength):
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverseFilter) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortFilter, setSortFilter] = useState(SortType.SortDefault);
+  const [isReverseFilter, setIsReverseFilter] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortFilter, isReverseFilter },
+  );
+
+  const resetFilter = () => {
+    setSortFilter(SortType.SortDefault);
+    setIsReverseFilter(false);
+  };
+
+  const handleSort = (sortField: SortType) => setSortFilter(sortField);
```

Comment:
```suggestion
  const handleSort = (sortField: SortType) => () => setSortFilter(sortField);
```
my suggestion was make function factory - which returns oher funciton. and just call in jsx